### DISCUSSION TITLE
Should we make preset upgradeable?

### DISCUSSION THREAD

**Discussion by martriay** [OpenZeppelin/cairo-contracts#878] [Polls]
Even though upgradeability is usually an opt in feature, Starknet's development speed and breaking changes make upgrades almost mandatory for some if not most contracts.

And even though the spirit of [our preset contracts](https://docs.openzeppelin.com/contracts-cairo/0.8.1/presets) is to be as minimal as possible and to redirect every customization request to [Wizard for Cairo](https://wizard.openzeppelin.com/cairo), the "ready-to-deploy" value proposition is not quite fulfilled if presets are not upgradeable. We received multiple requests for an upgradeable version of our presets, so I'm opening the discussion.

**Comment by martriay** [comment]
I believe presets should be upgradeable and anyone can opt out of them by generating a non upgradeable contract with Wizard.

Having both versions would defeat the purpose of minimality in terms of maintaining a single preset, opening the door to multiple preset flavors it's a slippery slope to maintain several others.

**Comment by milancermak** [comment]
I'm also in the upgradable-by-default camp. It's easy to remove upgradability, even in a deployed contract, but it's impossible to add.
